File: online_node_api/models/customer.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Customer extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
    }
  };
  Customer.init({
    f_cusID: {
      type:DataTypes.STRING,
      primaryKey:true,
      allowNull:false
    },
    f_cusName: DataTypes.STRING,
    f_cusDetail: DataTypes.STRING,
    f_cusStatus: DataTypes.INTEGER,
    f_cusNo: DataTypes.INTEGER
  }, {
    sequelize,
    modelName: 'Customer',
    tableName: 'sys_customer',
    timestamps:false
  });
  return Customer;
};

File: Flutter_Stock_Original/lib/models/customer.dart
class Customer {
  String f_cusID;
  String f_cusName;
  String f_cusDetail;
  int f_cusStatus;
  int f_cusNo;

  Customer(
      {required this.f_cusID,
      required this.f_cusName,
      required this.f_cusDetail,
      required this.f_cusStatus,
      required this.f_cusNo});

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
        f_cusID: json['f_cusID'],
        f_cusName: json['f_cusName'],
        f_cusDetail: json['f_cusDetail'],
        f_cusStatus: json['f_cusStatus'],
        f_cusNo: json['f_cusNo'] == null ? 0 : json['f_cusNo']);
  }
}
